src/main/java/pl/orki/hackathon/webapp/band/control/BandService.java

```diff
@@ -3,20 +3,37 @@
 import org.springframework.stereotype.Service;
 import pl.orki.hackathon.webapp.band.entity.Band;
 import pl.orki.hackathon.webapp.band.entity.BandRepository;
+import pl.orki.hackathon.webapp.city.entity.City;
+import pl.orki.hackathon.webapp.city.entity.CityRepository;
+import pl.orki.hackathon.webapp.genre.MusicGenre;
 import pl.orki.hackathon.webapp.user.entity.UserRepository;
 
 import javax.transaction.Transactional;
+import java.util.List;
 import java.util.Optional;
+import java.util.stream.Collectors;
 
 @Service
 public class BandService {
 
     private final BandRepository bandRepository;
     private final UserRepository userRepository;
+    private final CityRepository cityRepository;
 
-    public BandService(BandRepository bandRepository, UserRepository userRepository) {
+    public BandService(BandRepository bandRepository, UserRepository userRepository, CityRepository cityRepository) {
         this.bandRepository = bandRepository;
         this.userRepository = userRepository;
+        this.cityRepository = cityRepository;
+    }
+
+    public List<Band> getBandsByMusicGenresAndCities(List<String> genresNames, List<Long> citiesIds) {
+        List<City> cities = cityRepository.findAllById(citiesIds);
+        // TODO: it will be replaced by querying genres by ids
```

Comment:
TODOs lasts forever :P

Reply:
Not this one, after adding tests to these changes I will rewrite Genry to entity from enum as well ;)